pandas/core/frame.py

```diff
@@ -3385,15 +3385,9 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
                                                  sort_remaining=sort_remaining)
 
         elif isinstance(labels, MultiIndex):
-            from pandas.core.sorting import lexsort_indexer
-
-            # make sure that the axis is lexsorted to start
-            # if not we need to reconstruct to get the correct indexer
```

Comment:
so leave the ``is_lexsorted`` check in here; this is fixed in #15694 w/o this subtle things will break.